lib/simplifyExpression/simplify.js

```diff
@@ -23,14 +20,7 @@ function simplify(node, debug=false) {
     // removing parens isn't counted as a step, so try it here
     simplifiedNode = removeUnnecessaryParens(flattenOperands(node), true);
   }
-  // unflatten the node.
-  return unflatten(simplifiedNode);
-}
-
-// Unflattens a node so it is in the math.js style, by printing and parsing it
-// again
-function unflatten(node) {
```

Comment:
if we allow people to create their own trees, I think we might sometimes need to flatten - this is also why I flatten after every step using mathjs, even though we flatten at the beginning anyways and keep passing the same node around

it's possible we don't need it, but that's why my intuition is that we do